frontend/public/components/default-resource.jsx

```diff
@@ -19,27 +19,58 @@ import {
 const { common } = Cog.factory;
 const menuActions = [...common];
 
-const Header = props => <ListHeader>
-  <ColHead {...props} className="col-xs-6 col-sm-4" sortField="metadata.name">Name</ColHead>
-  <ColHead {...props} className="col-xs-6 col-sm-4" sortField="metadata.namespace">Namespace</ColHead>
-  <ColHead {...props} className="col-sm-4 hidden-xs" sortField="metadata.creationTimestamp">Created</ColHead>
-</ListHeader>;
+const computeColumnSizes = (columns, isResourceNamespaced) => {
+  const columnsNumber = isResourceNamespaced ? _.size(columns) + 1 : _.size(columns);
+  const columnsSize = Math.floor(12 / columnsNumber);
+  const fistColumnSize = 12 - (columnsSize * (columnsNumber-1));
+  return {
+    fistColumnSize,
+    columnsSize
+  };
+};
+
+const CustomHeader_ = props => {
+  if (!_.get(props, ['resourceTable', 'table'])) {
+    return null;
+  }
+  const { table, isNamespaced } = props.resourceTable;
+  const { fistColumnSize, columnsSize } = computeColumnSizes(table.columnDefinitions, isNamespaced);
+  const heads = _.map(table.columnDefinitions, (column, index) => {
+    const size = index === 0 ? fistColumnSize : columnsSize;
+    return <ColHead key={_.uniqueId()} className={`col-xs-${size}`}>{column.name}</ColHead>;
```

Comment:
I wouldn't use `_.uniqueId()` here. That will force it to recreate all the headers every redner. `index` would be better if there is nothing in the `column` object that's guaranteed to be unique.

Same comment for other uses of `_.uniqueId` for key below.

I think in the cases I see you could just use index since the table columns shouldn't change